packages/replay/src/replay.ts

```diff
@@ -372,15 +370,18 @@ export class ReplayContainer implements ReplayContainerInterface {
       return;
     }
 
+    // We can't move `_isEnabled` after awaiting a flush, otherwise we can
+    // enter into an infinite loop when `stop()` is called while flushing.
+    this._isEnabled = false;
```

Comment:
I moved this out of the try-catch, as otherwise this lead to weird race conditions. I guess by having this in the try-catch the timing semantics are slightly different. I had a bunch of cases where stop was then called twice at the exact same time, which was fixed by moving this out here 🤔

Reply:
Ah right, since try/catch inside of an async function is sugar on top of promises.